data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceGrpcService.java

```diff
@@ -37,16 +38,23 @@ public OTelTraceGrpcService(int bufferWriteTimeoutInMillis,
 
     @Override
     public void export(ExportTraceServiceRequest request, StreamObserver<ExportTraceServiceResponse> responseObserver) {
+        requestsReceivedCounter.increment();
+
+        if (Context.current().isCancelled()) {
```

Comment:
QUES: is it for client cancelling the request during server processing? I am not sure under what scenario/use case this could happen.

Reply:
Ah, I should put a comment here. This is to avoid processing gRPC requests which have passed their "deadline" - if the client says it will wait 500ms and the request isn't processed by the server until 600ms, the server can just drop the request because we assume the client has already moved on.

https://grpc.io/blog/deadlines/#checking-deadlines